src/main/java/com/openlattice/datastore/data/controllers/DataController.java

```diff
@@ -414,6 +418,101 @@ public Void clearEntitySet( UUID entitySetId ) {
+        authorizedPropertyTypes.put(
+                entitySetId,
+                authzHelper.getAuthorizedPropertyTypes( entitySetId, WRITE_PERMISSION )
+        );
+
+        /*
+         * 1 - get the entity neighbors
+         */
+
+        Map<UUID, List<NeighborEntityDetails>> result = searchService
+                .executeEntityNeighborSearch( entitySetId, ImmutableSet.of( entityKeyId ) );
+
+        if ( result != null && result.containsKey( entityKeyId ) ) {
+
+            /*
+             * 1.1 - collect authorized PropertyTypes per EntitySet and entities to clear per EntitySet
+             */
+
+            List<NeighborEntityDetails> neighbors = result.get( entityKeyId );
+            SetMultimap<UUID, UUID> neighborEntitiesMap = HashMultimap.create();
+
+            neighbors.parallelStream().forEach( neighbor -> {
+
+                /*
+                 * association EntitySet
+                 */
+                UUID associationEntitySetId = neighbor.getAssociationEntitySet().getId();
+                if ( !authorizedPropertyTypes.containsKey( associationEntitySetId ) ) {
+                    authorizedPropertyTypes.put(
+                            associationEntitySetId,
+                            authzHelper.getAuthorizedPropertyTypes( associationEntitySetId, WRITE_PERMISSION )
+                    );
+                }
+                UUID associationEntityKeyId = (UUID) neighbor.getAssociationDetails().get( ID_FQN ).iterator().next();
+                neighborEntitiesMap.put( associationEntitySetId, associationEntityKeyId);
+
+                /*
+                 * neighbor EntitySet
+                 */
+                if ( neighbor.getNeighborEntitySet().isPresent() ) {
+                    UUID neighborEntitySetId = neighbor.getNeighborEntitySet().get().getId();
+                    if ( !authorizedPropertyTypes.containsKey( neighborEntitySetId ) ) {
+                        authorizedPropertyTypes.put(
+                                neighborEntitySetId,
+                                authzHelper.getAuthorizedPropertyTypes( neighborEntitySetId, WRITE_PERMISSION )
+                        );
+                    }
+                    if ( neighbor.getNeighborDetails().isPresent() ) {
+                        SetMultimap<FullQualifiedName, Object> neighborDetails = neighbor.getNeighborDetails().get();
+                        UUID neighborEntityKeyId = (UUID) neighborDetails.get( ID_FQN ).iterator().next();
+                        neighborEntitiesMap.put( neighborEntitySetId, neighborEntityKeyId);
+                    }
+                }
+            } );
+
+            /*
+             * 1.2 - clear all neighbor and association entities
+             */
+
+            neighborEntitiesMap.keySet().forEach( neighborEntitySetId -> {
+                Set<UUID> neighborEntityKeyIds = neighborEntitiesMap.get( neighborEntitySetId );
+                dgm.clearEntities(
```

Comment:
Another major thing here is that graph edges are versioned so implementing this properly requires clearing versions in the edges as well as this is different from a a delete (clear only tombstones, while delete also deletes edges).